### ISSUE TITLE
Make the bot Recognize Signs

### ISSUE THREAD

**Issue by AndrewBuck** [opencog/opencog-to-minecraft#29]
The bot should remember the position of the sign (probably also which way it is facing) and also the text on the sign.  We can use the signs to tell the bot about important locations like its home and also convenience locations for things like "go harvest wood near the lake" (putting a sign by the lake that says 'lake' until the bot learns to recognize a lake).

**Comment by fredyvilla** [comment]
Another possibility to achieve something similar would be to use chat events.  You can teleport Bot, send a command to look at a coordinate, and then to remember the concept 'lake'

**Comment by AndrewBuck** [comment]
Yes, chat is something we definitely want to do, and may actually have to do first.  After I got the basics of locating signs in the code I went to investigate how the sign text is actually stored.  Apparently it is stored in a so called 'block entity' which is extra data stored separate from the block data itself.  Spockbot does not currently have any functions to read or access the block entity data, so this is going to be difficult if not impossible to implement for a while.

**Comment by Gjum** [comment]
SpockBotMC/SpockBot#214 implements block entities including signs.
